src/note_bariolage.lua

```diff
@@ -0,0 +1,88 @@
+
+function plugindef()
+  -- This function and the 'finaleplugin' namespace
+  -- are both reserved for the plug-in definition.
+  finaleplugin.Author = "Jacob Winkler"
+  finaleplugin.Copyright = "2022"
+  finaleplugin.Version = "1.0"
+  finaleplugin.Date = "3/18/2022"
+  finaleplugin.Notes = [[
+  USING THE 'BARIOLAGE' SCRIPT
+  
+This script creates bariolage-style notation where layers 1 and 2 interlock. It works well for material that has even-numbered beam groups like 4x 16th notes or 6x 16th notes (in compound meters). 32nd notes also work. Odd numbers of notes produce undesirable results.
+
+To use, create a suitable musical passage in layer 1, then run the script. The script does the following:
+- Duplicates layer 1 to layer 2.
+- Mutes playback of layer 2.
+- Iterates through the notes in layer 1. For even-numbered notes (i.e. the 2nd and 4th 16ths in a group of 4) it replaces the stem with a blank shape, effectively hiding it.
+- Any note in layer 1 that is the last note of a beamed group is hidden.
+- Iterates through the notes in layer 2 and changes the stems of the odd-numbered notes.
+- Any note in layer 2 that is the beginning of a beamed group is hidden.
+]]
+  return "Bariolage", "Bariolage", "Bariolage: Creates alternating layer pattern from layer 1. Doesn't play nicely with odd numbered groups!"
+end
+
+function layer_copy(src, dest)
```

Comment:
Could we convert this function to a library function? For instance `layer.copy`? I feel like future plugins could use this functionality.

If you use that name you'll need to create a new file under `src/library`, but there's nothing special you need to do. Here's the entire boilerplate you'll need.

```lua
--[[
$module Layer
]]

local layer = {}

-- your function here

return layer
```